src/prefect/task_server.py

```diff
@@ -100,87 +93,72 @@ async def stop(self):
         try:
             self._loops_task_group.cancel_scope.cancel()
         except Exception:
-            self._logger.exception(
-                "Exception encountered while shutting down", exc_info=True
-            )
+            logger.exception("Exception encountered while shutting down", exc_info=True)
 
     async def run_once(self):
         """Runs one iteration of the task server's polling cycle (used for testing)"""
         async with self._runs_task_group:
             await self._get_and_submit_task_runs()
 
-    async def _get_and_submit_task_runs(self):
-        if self.stopping:
-            return
-        runs_response = await self._get_pending_task_runs()
-        self._logger.debug(f"Found {len(runs_response)} task run(s)")
-        self.last_polled = pendulum.now("UTC")
-        await self._submit_pending_task_runs(task_run_response=runs_response)
-
-    async def _get_pending_task_runs(self) -> List[TaskRun]:
-        return await self._client.read_task_runs(
-            task_run_filter=TaskRunFilter(
-                state=dict(name=dict(any_=["Scheduled"])),
-                flow_run_id=dict(is_none=True),
-            ),
+    async def _subscribe_to_task_scheduling(self):
+        subscription = Subscription(TaskRun, "/task_runs/subscriptions/scheduled")
+        async for task_run in subscription:
+            logger.info(f"Received task run: {task_run.id} - {task_run.name}")
+            await self._submit_pending_task_run(task_run)
+
+    async def _submit_pending_task_run(self, task_run: TaskRun):
```

Comment:
This function is just the inner guts of the `_submit_pending_task_runs` loop, just intended to operate on one task run at a time; no other changes were made except that.